app/controllers/foreman_gutterball/api/v2/content_reports_controller.rb

```diff
@@ -4,22 +4,12 @@ module V2
       class ContentReportsController < ::Katello::Api::V2::ApiController
         api :GET, '/content_reports', 'List available reports'
         def index
-          service.available_reports
+          service.reports
         end
 
-        api :GET, '/content_reports/consumer_status', 'Consumer status reports'
-        def consumer_status
-          service.consumer_status
-        end
-
-        api :GET, '/content_reports/consumer_trend', 'Consumer trend reports'
-        def consumer_trend
-          service.consumer_trend
-        end
-
-        api :GET, '/content_reports/status_trend', 'Status trend reports'
-        def status_trend
-          service.status_trend
+        api :GET, '/content_reports/:id', 'Reports'
+        def show
+          service.reports(params[:id], request.query_string)
         end
```

Comment:
it feels pretty hacky hitting a different gutterball endpoint based on the existence of parameters.